src/main/java/pages/cartPage/shoppingCartTable/ShoppingCartTable.java

```diff
@@ -0,0 +1,150 @@
+        var price = getUnitPrice(rowNumber);
+        var imageUrl = rowContent.get(ShoppingCartHeader.Image.ordinal());
+
+        return ProductModel.builder()
+                .name(name)
+                .model(model)
+                .priceCurrency(priceCurrency)
+                .price(price)
+                .isOnSale(null)
+                .priceBeforeSale(null)
+                .isOutOfStock(null)
+                .imageUrl(imageUrl)
+                .build();
+    }
+
+    private String getUnitPriceWithSymbol(int rowNumber) {
+        var row = getRowContent(rowNumber);
+
+        return row.get(ShoppingCartHeader.UnitPrice.ordinal());
+    }
+
+    public Currency getCurrencyFromUnitPrice(int rowNumber) {
+        var row = getRowContent(rowNumber);
+        var currencySymbol = row.get(ShoppingCartHeader.UnitPrice.ordinal()).charAt(0);
+
+        return Currency.fromSymbol(String.valueOf(currencySymbol));
+    }
+
+    public BigDecimal getUnitPrice(int rowNumber) {
+        var unitPriceWithCurrency = getUnitPriceWithSymbol(rowNumber);
+        String unitPriceWithoutCurrency = unitPriceWithCurrency.substring(1);
+        return new BigDecimal(unitPriceWithoutCurrency);
+    }
+
+    public BigDecimal getUnitPrice(String productName) {
+        int rowNumber = getRowNumber(productName);
+        return getUnitPrice(rowNumber);
+    }
+
+    public BigDecimal getUnitPrice(ProductModel product) {
+        int rowNumber = getRowNumber(product.getName());
+        return getUnitPrice(rowNumber);
+    }
+
+    private String getTotalPriceWithCurrencyCharacter(int rowNumber) {
+        var row = getRowContent(rowNumber);
+        return row.get(ShoppingCartHeader.Total.ordinal());
+    }
+
+    public BigDecimal getTotalPrice(int rowNumber) {
+        String totalPriceWithCurrency = getTotalPriceWithCurrencyCharacter(rowNumber);
+        String totalPriceWithoutCurrency = totalPriceWithCurrency.substring(1);
+        return new BigDecimal(totalPriceWithoutCurrency);
+    }
+
+    public BigDecimal getTotalPrice(String productName) {
+        int rowNumber = getRowNumber(productName);
+        return getTotalPrice(rowNumber);
+    }
+
+    public BigDecimal getTotalPrice(ProductModel product) {
+        int rowNumber = getRowNumber(product.getName());
+        return getTotalPrice(rowNumber);
+    }
+
+    public void setQuantity(int rowNumber, int quantity) {
+        var row = getRow(rowNumber);
+        var input = row.findElements(By.tagName("td"))
+                .get(ShoppingCartHeader.Quantity.ordinal())
+                .findElement(quantityItemInput);
+
+        input.clear();
+        input.sendKeys(String.valueOf(quantity));
+    }
+
+    public void setQuantity(String productName, int quantity) {
+        int rowNumber = getRowNumber(productName);
+        setQuantity(rowNumber, quantity);
+    }
+
+    public void setQuantity(ProductModel product, int quantity) {
```

Comment:
In terms of our tests it might be good to also implement taking the current quantity, wdyf?

Reply:
Done
```java
    public int getQuantity(int rowNumber) {
        return Integer.parseInt(getQuantityInputValue(getRow(rowNumber)));
    }

    public int getQuantity(String productName) {
        int rowNumber = getRowNumber(productName);
        return getQuantity(rowNumber);
    }

    public int getQuantity(ProductModel product) {
        int rowNumber = getRowNumber(product.getName());
        return getQuantity(rowNumber);
    }

    private String getQuantityInputValue(WebElement cell) {
        return cell.findElements(quantityItemInput).getFirst().getAttribute("value");
    }
```

Reply:
Implemented in ShoppingCartTableRow
```java
    public int getQuantity() {
        return Integer.parseInt(getCells()
                .get(ShoppingCartHeader.QUANTITY.columnIndex)
                .findElement(quantityItemInput)
                .getAttribute("value"));
    }
```